vassal-app/src/main/java/VASSAL/counters/Footprint.java

```diff
@@ -720,6 +724,23 @@ public Command myKeyEvent(KeyStroke stroke) {
     return null;
   }
 
+  /**
+   * Displays a Bad Module Data warning if a movement trails trait is found inside of a rotation trait
+   */
+  public void requireNoOuterRotate() {
+    Decorator piece = getOuter();
+    while (piece != null) {
+      if ((piece instanceof FreeRotator) || (piece instanceof Pivot) || ((piece instanceof MatCargo) && (((MatCargo)piece).maintainRelativeFacing))) {
+        String name = getName();
+        if ((name == null) || name.isEmpty()) {
+          name = Resources.getString("Decorator.prototype");
+        }
+        ErrorDialog.dataWarning(new BadDataReport(Resources.getString("Decorator.trails_inside_rotate", getBaseDescription(), piece.getBaseDescription()), name));
+      }
```

Comment:
One of the constructors for BadDataReport takes an EditablePiece as an argument and includes the piece name in the report.

Reply:
That wouldn't handle the special case of a Prototype at least showing "Prototype" as the name (it couldn't actually show the real name of the prototype without adding a lot more "really bad code")

Reply:
Ok, true.